Ft #161255252 List all users

Tony-Ndichu opened the pull request:
#### What does this PR do?
This PR adds functionality for viewing all users and their profiles.

#### Description of Task to be completed?
Have the following endpoint working
- GET /api/profile/profiles/

#### How should this be manually tested?
- After cloning the repo, cd into it and run `python manage.py runserver`
- Register as a new user
- Activate account through email
- Login and copy the access token
- Add the access token in the header
- Hit the endpoint `GET /api/profile/all/`

#### What are the relevant pivotal tracker stories?
[#161255252](https://www.pivotaltracker.com/story/show/161255252)

#### Screenshots
- With authorization token: `GET /api/profile/all/`
<img width="1279" alt="screen shot 2018-11-06 at 10 14 43" src="https://user-images.githubusercontent.com/37045046/48048518-d5a84780-e1ac-11e8-84f4-e1de3717b2ca.png">



- Without authorization token: `GET /api/profile/all/`
<img width="1277" alt="screen shot 2018-11-06 at 10 15 33" src="https://user-images.githubusercontent.com/37045046/48048548-f40e4300-e1ac-11e8-8797-4996c57cca29.png">

- With authorization token: `GET /api/profile/me/`
<img width="1280" alt="screen shot 2018-11-06 at 13 31 20" src="https://user-images.githubusercontent.com/37045046/48058778-5b85bc00-e1c8-11e8-94c0-49a910c678d5.png">


- Without authorization token: `GET /api/profile/me/`
<img width="1275" alt="screen shot 2018-11-06 at 13 31 41" src="https://user-images.githubusercontent.com/37045046/48058792-65a7ba80-e1c8-11e8-8b6c-488c807b3e2d.png">

david-kironde-deel commented:
@Tony-Ndichu  what would be the most appropriate name for this endpoint? give me options.
- Add some screenshots.

Tony-Ndichu commented:
@Awesome94 ,how about '/all-users' ,  '/all-profiles' , '/all' , '/list' or '/profile-list'?

Commit:
feat(profiles): list all users' profiles and logged in user's profile

- ensure user is authenticated
- hit endpoint GET /api/profile/all/ to view all profiles
- hit endpoint GET /api/profile/me/ to view logged in user's profile

[Finishes #161255252]
